modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -617,39 +667,59 @@ object SeqexecEngine extends SeqexecConfiguration {
     )
   }
 
+  private def systemsBeingConfigured(st: EngineState): Set[Resource] =
+    st.sequences.values.filter(d => d.seq.status.isError || d.seq.status.isIdle).toList
+      .flatMap(s => s.seq.getSingleActionStates.filter(_._2.started).keys.toList
+        .mapFilter(s.seqGen.resourceAtCoords)
+      ).toSet
+
+  /*
+   * Resource in use = Resources used by running sequences, plus the systems that are being
+   * configured because a user commanded a manual configuration apply.
+   */
+  private def resourcesInUse(st: EngineState): Set[Resource] =
+    st.sequences.values.toList.mapFilter(s => s.seq.status.isRunning.option(s.seqGen.resources))
+      .foldK ++
+      systemsBeingConfigured(st)
+
+  /*
+   * Resources reserved by running queues.
+   */
+  private def resourcesReserved(failedSystems: Set[Resource], st: EngineState): Set[Resource] = {
+    def reserved(q: ExecutionQueue): Set[Resource] = q.queue.fproduct(st.sequences.get).collect{
+      case (_, Some(s)) if s.seq.status.isIdle &&
+        failedSystems.intersect(s.seqGen.resources).isEmpty => s.seqGen.resources
+    }.foldK
+
+    val runningQs = st.queues.values.filter(_.status(st).running)
+
+    runningQs.map(reserved).toList.foldK
```

Comment:
`foldK`?

Reply:
:sparkles: magic :sparkles:

Reply:
`List(a, b, c).combineAll` uses `Monoid` and gives you `a |+| b |+| c` whereas `.foldK` uses `MonoidK` and gives you `a <+> b <+> c`.

`MonoidK` is a universally-quantified monoid; i.e., `forAll a. Monoid[F[a]]`.  So a hand-wavy intuition is that the element type is unknown so you can't look at the elements. For example `Monoid[Option[Int]]` adds up any `Int`s it sees but `MonoidK[Option]` can't look at the elements so it uses `orElse`.

```scala
@ Option(1) |+| None |+| Option(2) 
res12: Option[Int] = Some(3)

@ Option(1) <+> None <+> Option(2) 
res13: Option[Int] = Some(1)
```